doc for setting up Service Catalog with Prometheus metrics

jboyd01 opened the pull request:
Documentation on how to view Service Catalog metrics and example YAML for deploying Prometheus to consume Catalog metrics.

pmorie commented on contrib/examples/prometheus/prometheus.yml:
```diff
@@ -0,0 +1,341 @@
+#
+# Prometheus will run uner the "prometheus" Service Account, create
```
typo: under

jboyd01 replied:
Thanks for the review @pmorie, typos matter.  All have bee addressed.

pmorie commented on contrib/examples/prometheus/README.md:
```diff
@@ -0,0 +1,87 @@
+Many metrics are not created and exposed until Service Catalog performs
+operations which would impact the metrics.  IE if you have no Service Brokers
+defined, there will be no metrics around class or plan count and likely no OSB
+Client operation metrics.  So before proceeding, its recommnded to you have a
```
recommended

pmorie replied:
also, "it's"

pmorie replied:
sorry, mom was an english teacher

pmorie commented on contrib/examples/prometheus/README.md:
```diff
@@ -0,0 +1,87 @@
+# Prometheus Metrics
+
+Service Catalog maintains metrics through the Prometheus client library and
+exposes them through the Prometheus http adapter at the /metrics endpoint.
```
This is the endpoint relative to the controller, right

jboyd01 replied:
Yes, correct.  I've updated this doc to be explicit.

MHBauer commented on contrib/examples/prometheus/prometheus.yml:
```diff
@@ -0,0 +1,341 @@
+#
+# Prometheus will run uner the "prometheus" Service Account, create
+# access to the necessary resources
```
What is "create access to the necessary resources"?

MHBauer replied:
I think maybe period for both as separate sentences. Still not sure why we would need the second sentence. It's rbac, it's all about access.

MHBauer commented on contrib/examples/prometheus/prometheus.yml:
```diff
@@ -0,0 +1,341 @@
+#
+# Prometheus will run under the "prometheus" Service Account, create
```
period at the comma. remove the rest.

MHBauer commented on contrib/examples/prometheus/README.md:
```diff
@@ -0,0 +1,87 @@
+facilitates analysis through a Web UI and powerful query language.
+
+Many metrics are not created and exposed until Service Catalog performs
+operations which would impact the metrics.  IE if you have no Service Brokers
```
IE -> In other words,

MHBauer commented on contrib/examples/prometheus/README.md:
```diff
@@ -0,0 +1,87 @@
+operations which would impact the metrics.  IE if you have no Service Brokers
+defined, there will be no metrics around class or plan count and likely no OSB
+Client operation metrics.  So before proceeding, it's recommended you have a
+Broker defined and created a Service Instance.
```
Not being familiar with prometheus, what does 'no metrics' look like? Does it mean everything is 0 or that until we do this prometheus won't even show what we've defined in the svccat?

MHBauer commented on contrib/examples/prometheus/README.md:
```diff
@@ -0,0 +1,87 @@
+$ kubectl get pods -l app=catalog-catalog-controller-manager -n catalog -o name | \
+    sed 's/^.*\///' | xargs -I{} kubectl port-forward {} -n catalog 8089:8080 &
+
+$ curl -s http://localhost:8089/metrics  | grep servicecatalog
```
this drops us into a pseudo shell of some sort?

jboyd01 replied:
The first command actually backgrounds a port forward allowing you to hit http://localhost:8089 and kube will forward the request to the catalog-controller pod's end point.

MHBauer commented on contrib/examples/prometheus/README.md:
```diff
@@ -0,0 +1,87 @@
+Kubernetes (ie [Prometheus
+Operator](https://github.com/coreos/prometheus-operator)) will pick up the
+Service Catalog metrics as long as it's looking for pods with the
+`prometheus.io/scrape` annotation.
```
this is really neat!

MHBauer commented on contrib/examples/prometheus/README.md:
```diff
@@ -0,0 +1,87 @@
+Prometheus.  [This YAML](prometheus.yml) creates a Prometheus instance
+preconfigured to gather Kubernetes platform and node metrics.  If you deploy the
+Service Catalog Controller Manager via Helm with the optional
+`enablePrometheusScrape` parameter set to true (see
```
We should probably write the exact `helm set` here in addition to the general descriptive guideline.

jboyd01 commented:
Thanks for the review @MHBauer.  I fixed the over zealous comments in the YAML file and addressed the other issues.  I meant this more as a developer/experimenter guide with the plan of adding real/better metrics documentation.  Once we have more Catalog metrics I hope we can call out specific items to monitor in production and perhaps even suggestions on what to alert off from.

pmorie commented:
Thanks for this, @jboyd01!
